odoo/orm/models.py

```diff
@@ -3164,12 +3167,15 @@ def fetch(self, field_names: Collection[str] | None = None) -> None:
 
         # fetch the fields
         fetched = self._fetch_query(query, fields_to_fetch)
+        env = self.env
+        if not env.su and type(self)._check_access is BaseModel._check_access:
```

Comment:
Not a huge fan of `type(self)._check_access is BaseModel._check_access`. it still needed? _search should always have the same behavior than `_check_access` or it can be exploit by the domain anyway.

Reply:
As you write they should. I have 2 cases where it is not exactly true (although it is closer than when I started writing a POC for this PR).
- attachment: https://github.com/odoo/odoo/pull/219015 (done)
- mail: at least https://github.com/odoo/odoo/pull/232245 or better https://github.com/odoo/odoo/pull/221906
- spreadsheet: https://github.com/odoo/enterprise/pull/97596

I have rebased and pushed a commit to remove that check. I think we could go with it and decide what to do with PRs referenced here.